extensions/ql-vscode/src/quickEvalCodeLensProvider.ts

```diff
@@ -0,0 +1,39 @@
+import {
+  CodeLensProvider,
+  TextDocument,
+  CodeLens, 
+  Command,
+  Range
+} from 'vscode';
+  
+class QuickEvalCodeLensProvider implements CodeLensProvider {
+  async provideCodeLenses(document: TextDocument): Promise<CodeLens[]> {
+
+    const codeLenses: CodeLens[] = [];
+
+    for (let index = 0; index < document.lineCount; index++) {
+      const textLine = document.lineAt(index);
+      // Match a predicate signature, including predicate name, parameter list, and opening brace.
+      const regex = new RegExp(/(\w+)\s*\(\s*.*(?:,\s*)*\)\s*\{/);
+      const matches = textLine.text.match(regex);
+
+      if (matches) {
```

Comment:
Ensure that commented code removes the code lens. It's too tricky to do in a single regex, so I think doing another check is ok.

```suggestion
      if (matches && textLine.text.search(/^\s*\//)) {
```

Reply:
We had an illuminating discussion offline about why this works (`search` returns 0 when it matches the start of the line and -1 on mismatch, which JS converts into `false` and `true` respectively). I suggested a slightly more verbose but arguably simpler use of `test` instead -- hopefully that's still captured the meaning you suggested here!

Reply:
A downside with the suggestion is that the regex is evaluated twice for every line, even if there is no initial match. The original suggestion only ran the second regex _if_ the first one matched. I'm not sure if the extra overhead would be noticeable, but this code is run after every file change (in a background thread).

Reply:
That's a good point. We can keep using `!test` rather than `search`, but keep it inline instead of in its own variable.
So `if (matches && /.../.test(textLine.text)`

Reply:
Yep....that makes sense.

Reply:
Made this change. Sorry it took a bit. I forgot I had resolved the `CHANGELOG` conflict from here but I hadn't updated my local branch yet. Had to get that squared away. Does it look better?